app/validators/action_kit_fields.rb

```diff
@@ -0,0 +1,94 @@
+# The <tt>ActionKitFields</tt> class is a Rails <tt>ActiveModel::Validator<tt> class
+# for validating a model's +name+ field against ActionKits +action+ and +user+
+# tables.
+#
+# Valid field names must be whitelisted ( see <tt>ACTIONKIT_FIELDS_WHITELIST</tt> ),
+# or they must be prefixed with +user_+ or +action_+. All field names must not
+# have spaces or dashes.
+#
+# See also https://act.sumofus.org/docs/manual/api/rest/actionprocessing.html for detailed documentation
+# on ActionKit's action processing.
+#
+# ==== Valid Field Names
+#
+#   address1
+#   state
+#   action_bar
+#   action_foo_bar
+#   user_foo
+#
+# ==== Invalid Field Names
+#
+#   address_action
+#   foo-bar
+#   foobar
+#   action_foo bar
+#
+
+class ActionKitFields < ActiveModel::Validator
+  VALID_CHARS_RE = /^[0-9a-z_]+$/
+
+  VALID_PREFIX_RE = /^(action|user)\_[0-9a-z_]+/
+
+  ACTIONKIT_FIELDS_WHITELIST = %w{
+    address1
+    address2
+    city
+    country
+    email
+    first_name
+    home_phone
+    last_name
+    middle_name
+    mobile_phone
+    name
+    password
+    phone
+    plus4
+    postal
+    prefix
+    region
+    state
+    subscription_status
+    suffix
+    zip
+  }
```

Comment:
sorry, my previous comment got cut off.  @paulaferris, you don't want me to explain here how I think it should work?  how should I convey that information?  At the very least, campaigners should not have to prefix custom fields with action_ ... AS already does this for them and almost none of the campaigners know that the prefix gets added behind the scenes, so I think that will be confusing in testing.  We currently don't use any user_ fields ... we would like to someday, but that can safely be dropped ... so action_ can just be prepended to all custom fields.   I am concerned that if the process for creating a custom field is the same as just adding an AK field, campaigners could do it by accident ... for instance, a campaigner wants to add address1, but instead creates a custom field called street_address. The field subscription_status should not be exposed to the campaigner or the user.  I have further validation requirements for checkboxes, but those can wait ... the others I think are simple.